projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/api/meta_job.py

```diff
@@ -22,18 +28,25 @@ class SingleJob:
 @dataclass
 class MetaJob(SingleJob):
     """
-    TODO
+    This class represents a DAG Job, which is a single job itself and consists of single jobs - the orchestrated ones.
+
+    :param jobs: A list of the orchestrated jobs
     """
 
     jobs: List[SingleJob] = field(default_factory=list)
 
 
 class IMetaJobInput(abc.ABC):
+    """
+    This class is responsible for the DAG job run.
+    """
+
```

Comment:
What's the commonality between `IMetaJobInput` and `IJobInput`?

Reply:
I don't think there is anything in common.

Reply:
nit: This may be a separate design topic to this PR.

Looking at the names, they are both job input interfaces and one is for a DAG, and the other one is for a composite data job. 

From _API perspective_, however, those are now used differently -> the `JobInput` object is **managed by VDK**, and provided to the user via run method parameters; while `DagJobInput` instance is expected to be managed (created and handled) by the user.
And that's why, IDagJobInput does not extend IJobInput, from _OOP perspective_.

So now, `file_based_step.py` supports:
```
@staticmethod
    def run_python_step(step: Step, job_input: IJobInput) -> bool:
```

And, as a user extending VDK, I would appreciate extensibility of file steps of type python. 😊 
So that, having 
```
def run(dag_job_input: IDagJobInput):
```
is also a valid step of type Python.

Reply:
I think I understand your point. We can take this topic to a separate discussion - share more thoughts about the cons of the current approach and the benefits of having such common interface and the implementation details around this. should we open a GitHub issue about it? What do you think?